lib/nodejs/file-cache.js

```diff
@@ -37,19 +48,33 @@ function _FileCache( ) {
         // and have it already cached, if so, return the
         // already cached file.
         for ( var i = 0; i < this.files.length; i++ ) {
-            if( this.files[ i ].path == path ) {    
+            if( this.files[ i ].path == path ) {
                 global.log( 'serving from cache: ' + path, 2 );
                 callback( this.files[ i ] );
                 return;
             }
         }
 
         // if got here, have no record of this file yet.
-        var datatype = this.getDataType( path );
-        var file = fs.readFileSync( path );
+        var ext = path.substr( path.lastIndexOf( '.' ) + 1 ).toLowerCase();
+        var datatype = this.getDataType( ext );
+        var file = ext !== 'html' ? fs.readFileSync( path) : fs.readFileSync( path, datatype );
         var stats = fs.statSync( path );
 
         if ( file ) {
+            //  if the file is html replace all autoversion entries with
+            //    path appended by mtime parameter before adding to cache
+            if ( ext === 'html' ) {
+                var dirPath = path.substring( 0, path.lastIndexOf( '\\' ) + 1 );
+                var fileStr = file.replace(/\[autoversion\((.*?)\)\]/g, function(match) {
+                        var autoStr = '[autoversion(';
+                        match = match.substring(autoStr.length, match.length - 2);
+                        var mtime = FileCache.getLastModifiedTime(dirPath+match);
```

Comment:
`getLastModifiedTime` doesn't refer to any `_FileCache` members, so you could make it a local function and avoid this reference to the `FileCache` global.